### DISCUSSION TITLE
librdkafka client different behaviour 1.6.0 vs 1.8.x  & 1.9.0

### DISCUSSION THREAD

**Discussion by spstream** [edenhill/librdkafka#3771] [Q&A]
Hello !  I have observed the strange situation with the librdkafka different versions.

rk = rd_kafka_new(RD_KAFKA_CONSUMER, conf, errstr, sizeof(errstr));
version 1.6.0 => Works fine
version   1.8.2 / 1.9.0  =>  Fails with rts 114 "Attempt to access item beyond bounds of memory"

scenario - using the supplied  example program  consumer.c / producer.c  and calling that from a cobol program.   with all versions, producer.c call works fine when calling from cobol.  however consumer.c call works only with librdkafka 1.6.0 version, while for all greater version listed above it fails with "Attempt to access item beyond bounds of memory"

any body experienced similar situation ?  any idea what could be wrong ? 

Additional info:  
1: This is 32 bit build of librdkafka
2: Direct "C" executable for all version works fine.
3: rts 114 issue arise when consumer "C"  program called from cobol for version > 1.6.0.

**Comment by edenhill** [comment]
I think it might be better to create an issue for this and fill out the checklist (in the issue template) so that we know what platform and config is being used.

Also try running it in a debugger to understand what is happening.
